### PR TITLE
[PRD-6165] Reset sheet row on new sheet

### PR DESCRIPTION
**Pull request by befc**
@pentaho/tatooine_dev please review

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 10m 41s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl engine/core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.reporting.engine.classic.core.RotationTest.testHandleRotatedTextHTML</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

  
Tests run: **823**, Failures: **1**, Skipped: **2** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/reporting/job/PR-1679/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by befc** [comment]
The failing test is currently failing in other [recently opened PR](https://github.com/pentaho/pentaho-reporting/pull/1678) and [latest QAT](https://thanos.orl.eng.hitachivantara.com/job/10.3/job/suite-release/41/testReport/)

File: engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/output/table/xls/helper/ExcelPrinter.java [hunk 1]
```diff
@@ -124,6 +124,7 @@ public void print( final LogicalPageKey logicalPageKey, final LogicalPageBox log
 
     if ( sheet == null ) {
       addNewSheet( logicalPage, contentProducer, excelTableContentProducer );
+      sheetRow = 0;
```
Comment (mladenmilev) [inline engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/output/table/xls/helper/ExcelPrinter.java:127]: please add a comment referring to the BUG causing this change, similar line 122.

**Pull request merged**